### PR TITLE
HTML Endpoint

### PR DESCRIPTION
**Pull request by michael-r-may**
Added the ability to be able to return html instead of json so we can demonstrate the bff without needing a native client.

### CODE REVIEW

File: src/main/kotlin/com/romainpiel/catsanddogs/ScheduleRepository.kt [hunk 1]
```diff
@@ -18,6 +22,21 @@ enum class Conference(val rawValue: String) {
...
 
+fun Single<List<Item>>.toHtml(): Single<String> {
+    return this.map { it.map { it }.fold("", { acc, item ->
+        val title = item.title
+        val subtitle = item.subtitle
+        val time = item.time
+        val date = item.date
+
+        acc + "<hr><p>" +
+                "<b>$title</b><br>" +
```
Comment (romainpiel) [inline src/main/kotlin/com/romainpiel/catsanddogs/ScheduleRepository.kt]: ${item.title}

File: src/main/kotlin/com/romainpiel/catsanddogs/ScheduleRepository.kt [hunk 2]
```diff
@@ -18,6 +22,21 @@ enum class Conference(val rawValue: String) {
     }
 }
 
+fun Single<List<Item>>.toHtml(): Single<String> {
+    return this.map { it.map { it }.fold("", { acc, item ->
```
Comment (romainpiel) [inline src/main/kotlin/com/romainpiel/catsanddogs/ScheduleRepository.kt]: .flatMapIterable { it }

File: src/main/kotlin/com/romainpiel/catsanddogs/ScheduleRepository.kt [hunk 3]
```diff
@@ -41,6 +60,21 @@ class ScheduleRepository {
...
+    fun schedule(from: OffsetDateTime, locale: Locale, conference: Conference): Single<String> {
+        return this.scheduleList(from, locale, conference).toJson(gson)
+    }
+
+    fun scheduleHtml(from: OffsetDateTime, locale: Locale, conference: Conference): Single<String> {
+        return this.scheduleList(from, locale, conference).toHtml()
     }
 }
+
+fun offsetDateTime(queryParameters: ValuesMap): OffsetDateTime {
```
Comment (romainpiel) [inline src/main/kotlin/com/romainpiel/catsanddogs/ScheduleRepository.kt]: queryParameters.from()


```kotlin
fun ValuesMap. from() {
     val fromStr: String? = this["from"]
     val from = fromStr?.let { OffsetDateTime.parse(it) }
     val safeFrom = from ?: OffsetDateTime.MIN
 
     return safeFrom
}
```

File: src/main/kotlin/com/romainpiel/catsanddogs/ScheduleRepository.kt [hunk 4]
```diff
@@ -41,6 +60,21 @@ class ScheduleRepository {
...
                 .toList()
-                .toJson(gson)
+    }
+
+    fun schedule(from: OffsetDateTime, locale: Locale, conference: Conference): Single<String> {
+        return this.scheduleList(from, locale, conference).toJson(gson)
+    }
+
+    fun scheduleHtml(from: OffsetDateTime, locale: Locale, conference: Conference): Single<String> {
+        return this.scheduleList(from, locale, conference).toHtml()
```
Comment (romainpiel) [inline src/main/kotlin/com/romainpiel/catsanddogs/ScheduleRepository.kt]:
```kotlin
fun scheduleHtml(from: OffsetDateTime, locale: Locale, conference: Conference) =
   this.scheduleList(from, locale, conference).toHtml()
```

**Comment by michael-r-may** [comment]
Hey @RomainPiel I've made all those changes except the `.flatMapIterable { it }` which I cannot seem to make happen. Feel free to have a go. Or merge. Or both! :-)

**Comment by romainpiel** [comment]
so it happened that you didn't need the `map { it }` at the end. looking good, merge it!

**Pull request merged**